sklearn/_display_estimator.py

```diff
@@ -0,0 +1,308 @@
+from sklearn._config import config_context
+from contextlib import closing
+from io import StringIO
+import uuid
+
+
+class _EstHTMLBlock:
+    """HTML Representation of Estimator
+
+    Parameters
+    ----------
+    type : {'serial', 'parallel', 'single'}
+        Type of HTML block
+
+    estimators : list of estimators or _EstHTMLBlock
+                or a single estimator
+        If type is in ('parallel', 'serial'), then `estimators` is a list of
+        estimators.
+        If type == 'single', then `estimators` is a single estimator.
+
+    names : list of str
+        If type in ('parallel', 'serial'), then `names` corresponds to
+        estimators
+        If type is 'single', then `names` is a single string corresponding to
+        the single estimator.
+
+    name_details : list of str, str, or None, default=None
+        If type == 'parallel', then `name_details` corresponds to `names`.
+        If type == 'single', then `name_details` is a single string
+        corresponding to the single estimator.
+        `name_details` is not used when type == 'single'.
+
+    dash_wrapped : bool, default=True
```

Comment:
Elsewhere here the naming reflects the semantics. Here it reflects the aesthetics. I wonder if we could make it more semantic.